### PR TITLE
Improve backburner job name sanitization

### PR DESCRIPTION
**Pull request by jpbrault**
JIRA: SMOK-51684
Shotgun can submit backburner jobs with a job name that is too long, error on submission

The maximum job name is 100 characters. However, we want to limit a bit lower
than than since duplicate job name will have a _### appended to the end of the
job name.

Create a method in the enging to do that sanitization to avoid code duplication
and support both a prefix and suffix to the job name that won't be ellipsed
unless we overflow too much.

Address some pylint warning/refactor issues.

### CODE REVIEW

File: engine.py [hunk 1]
```diff
@@ -1369,6 +1417,41 @@ def get_backburner_job_completion(self):
                 )
         return (completion_handling, completion_handling_delay)
 
+    @staticmethod
+    def sanitize_backburner_job_name(job_name, job_prefix="", job_suffix=""):
+        """
+        Sanitize a job name so backburner will accept it. Will replace undesired character with '_'
+        and will ellips the string if too long.
```
Comment (morinm-adsk) [inline engine.py]: ellips

File: engine.py [hunk 2]
```diff
@@ -1369,6 +1417,41 @@ def get_backburner_job_completion(self):
                 )
         return (completion_handling, completion_handling_delay)
 
+    @staticmethod
+    def sanitize_backburner_job_name(job_name, job_prefix="", job_suffix=""):
+        """
+        Sanitize a job name so backburner will accept it. Will replace undesired character with '_'
```
Comment (morinm-adsk) [inline engine.py]: character -> characters

File: python/tk_flame/transcoder.py [hunk 3]
```diff
@@ -83,9 +83,11 @@ def preExportAsset(self, info, userData, *args, **kwargs):
                 pass
 
             def postExportAsset(self, info, userData, *args, **kwargs):
+                del args, kwargs # Unused necessary parameters
```
Comment (morinm-adsk) [inline python/tk_flame/transcoder.py]: If an object is passed by reference (dict), will that delete the original object?

Comment (jpbrault) [reply python/tk_flame/transcoder.py]: No. I had the same worries

https://onlinegdb.com/B1sUWntpN

**Pull request merged**